### PR TITLE
Move the config for ci-search under core-services

### PR DESCRIPTION
**Pull request by petr-muller**
...so it can be deployed with applyconfig and is covered by a dry-run job and automated deploy postsubmit.

/cc @stevekuznetsov @smarterclayton @bbguimaraes @droslean @hongkailiu

### CODE REVIEW

File: cluster/ci/config/prow/plugins.yaml [hunk 1]
```diff
@@ -1240,6 +1240,10 @@ config_updater:
     cluster/ci/monitoring/build_cop.json:
       name: grafana-dashboard-build-cop
       namespace: prow-monitoring
+    core-services/ci-search/_config.yaml:
+      name: job-config
+      namespace: ci-search
+      key: config.yaml
```
Comment (petr-muller) [inline cluster/ci/config/prow/plugins.yaml]: We do not need to do this separately, the CM already exists and the content is up2date.

**Comment by droslean** [comment]
We need to manually update the `config_updater` rbacs.. I feels that we hit that a lot lately.. Shall we consider to automate this?

**Comment by petr-muller** [comment]
> We need to manually update the `config_updater` rbacs.. I feels that we hit that a lot lately.. Shall we consider to automate this?

I'm doing that in https://github.com/openshift/release/pull/4478, not that much hassle and it makes me re-think if what permissions I'm giving. But I would not mind an automated help.

**Pull request merged**